fix compilation error on GHC-8.8

msakai opened the pull request:
MonadFail proposal (MFP) removed `fail` from `Monad` class in GHC-8.8 (base-4.13).

msakai committed:
fix compilation error on GHC-8.8

MonadFail proposal (MFP) removed 'fail' from Monad class in GHC-8.8 (base-4.13).

msakai committed:
add GHC-8.8 to Travis-CI build matrix

philderbeast commented:
Thanks @msakai. The travis checks show we have hlint failures.

```
Test suite hlint: RUNNING...
src/Data/UnitsOfMeasure/Internal.hs:4:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE KindSignatures #-}
Perhaps you should remove it.
Note: Extension KindSignatures is implied by TypeFamilies
src/Data/UnitsOfMeasure/Singleton.hs:3:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE KindSignatures #-}
Perhaps you should remove it.
Note: Extension KindSignatures is implied by TypeFamilies
2 hints
Test suite hlint: FAIL
```

msakai committed:
Remove {-# LANGUAGE KindSignatures #-} as suggested by hlint

KindSignatures is implied by TypeFamilies.

msakai commented:
Thanks. I'll work on that.

msakai commented:
I added GHC-8.8.3 to Travis-CI's build matrix and fixed the hlint failures.

Although, the build succeeded, haddock issued the following warning.

```
haddock: panic! (the 'impossible' happened)
  (GHC version 8.8.3 for x86_64-unknown-linux):
	mkPluginUsage: file not found
Data.UnitsOfMeasure.Plugin /tmp/.haddock-16403/Data/UnitsOfMeasure/Plugin.o
Call stack:
    CallStack (from HasCallStack):
      callStackDoc, called at compiler/utils/Outputable.hs:1159:37 in ghc:Outputable
      pprPanic, called at compiler/deSugar/DsUsage.hs:235:15 in ghc:DsUsage
Please report this as a GHC bug:  https://www.haskell.org/ghc/reportabug
Warning: Failed to build documentation for uom-plugin-0.3.0.0.
```

It seems the problem was already reported at https://gitlab.haskell.org/ghc/ghc/issues/16093 and I think it's better to handle this issue separately from this PR.

philderbeast commented:
@msakai are you aware that this plugin builds but is not of use with versions of ghc > 8.2.2? This is why we don't run the unit testsuite. Please see #43 for more detail.

https://github.com/adamgundry/uom-plugin/blob/8a27dc794a0f0e4955224ee739e21961a8fe4411/uom-plugin/uom-plugin.cabal#L100-L101

philderbeast commented:
@msakai thanks for logging #66 as a separate issue.

msakai commented:
> are you aware that this plugin builds but is not of use with versions of ghc > 8.2.2?

Thanks. I hadn't noticed it yet.
I recently read the paper "A Typechecker Plugin for Units of Measure" and was just trying out the plugin.

philderbeast commented:
I've added a warning to the readme about #43.
